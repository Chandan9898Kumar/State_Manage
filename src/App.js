import { BrowserRouter, Route, Routes } from "react-router-dom";
import React, { lazy, Suspense } from "react";
import "./App.css";

// import MainHeader from './StateManageComponent/PreserveState/MainComponent'
// import Parent from "./StateManageComponent/SharingStateBetweenTwoComponent/ParentCom";
// import HeaderNav from "./StateManageComponent/Header";
// import PracticeOne from "./StateManageComponent/StateManageFirst";

//                 Instead using Lazy

// Here we are using lazy function instead of directly importing component.
const HeaderNav = lazy(() => import("./StateManageComponent/Header"));
const PracticeOne = lazy(() =>
  import("./StateManageComponent/StateManageFirst")
);
const Parent = lazy(() =>
  import("./StateManageComponent/SharingStateBetweenTwoComponent/ParentCom")
);
const MainHeader = lazy(() =>
  import("./StateManageComponent/PreserveState/MainComponent")
);
function App() {
  return (
    <BrowserRouter>
      <Suspense fallback={<div className="loader"></div>}>
        <Routes>
          <Route exact path="/" element={<HeaderNav />} />
          <Route exact path="/StateOne" element={<PracticeOne />} />
          <Route
            exact
            path="/SharingStateBetweentwoComp"
            element={<Parent />}
          />
          <Route
            exact
            path="/PreserveState"
            element={<MainHeader />}
          />
        </Routes>
      </Suspense>
    </BrowserRouter>
  );
}

export default App;
