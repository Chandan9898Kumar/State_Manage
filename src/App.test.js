import { render, screen } from "@testing-library/react";
import App from "./App";

describe("Render App", () => {
  test("Match Snapshot", () => {
    const { container } = render(<App />);
    expect(container).toMatchSnapshot();
  });

  test("App To be defined", () => {
    const { container } = render(<App />);
    expect(container).toBeDefined();
  });
});
