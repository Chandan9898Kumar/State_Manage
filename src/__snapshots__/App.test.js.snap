// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` renders learn react link 1`] = `
<div>
  <div
    class="loader"
  />
</div>
`;

exports[`Render App App To be defined 1`] = `
<div>
  <div>
    <ul
      class="nav"
    >
      <li>
        <a
          class=""
          href="/StateOne"
          style="position: absolute; left: 20px; width: 100px; height: 120px; color: white;"
        >
          State One
        </a>
      </li>
      <br />
    </ul>
  </div>
</div>
`;

exports[`Render App Match Snapshot 1`] = `
<div>
  <div
    class="loader"
  />
</div>
`;

exports[`Render App Match Snapshot 2`] = `
<div>
  <div>
    <ul
      class="nav"
    >
      <li>
        <a
          class=""
          href="/StateOne"
          style="position: absolute; left: 20px; width: 100px; height: 120px; color: white;"
        >
          State One
        </a>
      </li>
      <br />
    </ul>
  </div>
</div>
`;

exports[`Render App renders learn react link 1`] = `
<div>
  <div
    class="loader"
  />
</div>
`;

exports[`renders learn react link 1`] = `
<div>
  <div
    class="loader"
  />
</div>
`;
